internal/expr/typeinfo.go

```diff
@@ -109,6 +112,67 @@ func getTypeInfo(value any) (typeInfo, error) {
 	return typeInfo, nil
 }
 
+// fieldIsStruct checks if the field type is an embedded/nested struct or
+// pointer to an embedded/nested struct.
+// If a pointer to the field type implementes Scanner then it is not a nested
+// struct.
+func fieldIsStruct(field reflect.StructField) bool {
+	ft := field.Type
+	k := ft.Kind()
+	// Check if it is a
+	return (k == reflect.Struct && !reflect.PointerTo(ft).Implements(scannerInterface)) ||
+		(k == reflect.Pointer && ft.Elem().Kind() == reflect.Struct && !ft.Implements(scannerInterface))
+}
+
+// getStructFields returns relevent reflection information about all struct
+// fields included nested/embedded fields.
+// The caller is required to check that t is a struct.
```

Comment:
* Can we say `structType` instead of `t`?

Additional [Nit]: Can we say that `structType` is required to be a struct instead of talking about the caller checking it?